addons/binding/org.openhab.binding.vitotronic/src/main/java/org/openhab/binding/vitotronic/handler/VitotronicThingHandler.java

```diff
@@ -0,0 +1,181 @@
+ * 
+ * @author Stefan Andres - Initial contribution
+ */
+
+public class VitotronicThingHandler extends BaseThingHandler implements VitotronicThingListner {
+
+    private Logger logger = LoggerFactory.getLogger(VitotronicThingHandler.class);
+    
+    private int pollingBridge = 60; // every seconds ask for Bridge
+    VitotronicBridgeHandler bridgeHandler;
+
+
+	public VitotronicThingHandler(Thing thing) {
+		super(thing);
+	}
+	
+	@Override
+	public void initialize() {
+		logger.debug("Thing Handler for {} started", getThing().getUID().getId() );
+		bridgeOnlineWatchdog();
+	}
+	
+	@Override
+	public void dispose() {
+		logger.debug("Thing Handler for {} stop", getThing().getUID().getId() );
+		unregisterThingListener();
+	}
+
+	@Override
+	public void handleCommand(ChannelUID channelUID, Command command) {
+		logger.trace("Handle command for channel '{}' command '{}'", channelUID.getId(), command.toString());
+		bridgeHandler.updateChannel(getThing().getUID().getId(), channelUID.getId(), command.toString());
+     
+	}
+	
+	private void bridgeOnlineWatchdog() {
+		Runnable runnable = new Runnable() {
+			public void run() {
+				try {
+					bridgeHandler = getBridgeHandler();
+					if (bridgeHandler != null) {
+						updateStatus(bridgeHandler.getStatus());
+						registerThingListener();
+					} else {
+						updateStatus(ThingStatus.OFFLINE);
+						logger.debug("Bridge not found.");
+					}
+				} catch (Exception e) {
+					logger.debug("Exception occurred during execution: {}", e.getMessage(), e);
+				}
+			  }
+			};
+		@SuppressWarnings("unused")
+		ScheduledFuture<?> pollingJob = scheduler.scheduleAtFixedRate(runnable, 0, pollingBridge, TimeUnit.SECONDS);
+		}
+	
+
+	private synchronized  VitotronicBridgeHandler getBridgeHandler() {
+		
+		VitotronicBridgeHandler bridgeHandler = null;
+		
+		Bridge bridge = getBridge();
+		  if (bridge == null) {
+		     logger.debug("Required bridge not defined for device {}.");
+		     return null;
+		  }
+		  ThingHandler handler = bridge.getHandler();
+		  if (handler instanceof VitotronicBridgeHandler) {
+		      bridgeHandler = (VitotronicBridgeHandler) handler;
+		  } else {
+			  logger.debug("No available bridge handler found for bridge: {} .", bridge.getUID());
+			  bridgeHandler = null;
+		  }	
+		return bridgeHandler;
+	}
+	
+	
+	
+	private void registerThingListener(){
+		bridgeHandler.registerThingListener(this);
+	}
+	
+	private void unregisterThingListener(){
+		bridgeHandler.unregisterThingListener(this);
+		
+	}
+	
+	// Storing channel data -> only update changes 
+	private Map<String,String> channelData = new HashMap<String,String>();
+	
+	void setChannelValue(String channelId, String value) {
+		Channel channel = getThing().getChannel(channelId);
+		if (channel == null) {
+			logger.warn("Cannel '{}:{}' not implemented", getThing().getUID().getId(), channelId);
+			return;
+		} 
+		
+		// Check if data is changed
+		
+		if (channelData.get(channelId) !=null) {

```

Comment:
if the data is up-to-date, you should actually not filter it, but also send an update if the value has not changed